bin/crew

```diff
@@ -553,77 +569,110 @@ def download
+            FileUtils.cp cachefile, CREW_BREW_DIR, verbose: @fileutils_verbose
+            puts "Archive found in cache".lightgreen
+            return {source: source, filename: filename}
           else
-          puts 'Cached archive checksum mismatch. :/ Will download.'.lightred
+            puts 'Cached archive checksum mismatch. :/ Will download.'.lightred
+          end
+        puts 'Cannot find cached archive. :/ Will download.'.lightred
         end
       end
-    end
-
-    system "#{CURL} --retry 3 -#{@verbose}#LC - --insecure \'#{url}\' --output #{filename}"
-
-    abort 'Checksum mismatch. :/ Try again.'.lightred unless
-      Digest::SHA256.hexdigest( File.read(filename) ) == sha256sum
-    puts "Archive downloaded".lightgreen
+      # Download file if not cached
+      system "#{CURL} --retry 3 -#{@verbose}#LC - --insecure \'#{url}\' --output #{filename}"
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless
+        Digest::SHA256.hexdigest( File.read(filename) ) == sha256sum
+      puts "Archive downloaded".lightgreen
+      # Stow file in cache if requested
+      if CREW_CACHE_ENABLED
+        puts 'Caching downloaded archive'.lightgreen
+        FileUtils.cp filename, CREW_CACHE_DIR, verbose: @fileutils_verbose
+      end
 
-    if ENV['CREW_CACHE_OPT']
-      puts 'Caching downloaded archive'.lightgreen
-      FileUtils.cp filename, CREW_CACHE_DIR, verbose: @fileutils_verbose
+    # Sources that download with git
+    when /\.git$/i
+      # Recall repository from cache if requested
+      if CREW_CACHE_ENABLED
+        cachefile = CREW_CACHE_DIR + '/' + filename + '.tar.xz'
+        if File.file?(cachefile)
+          if system "sha256sum -c #{cachefile}.sha256"
+            FileUtils.mkdir @extract_dir
+            system "tar x#{@verbose}f #{cachefile} -C #{@extract_dir}"
+            return {source: source, filename: filename}
+          else
+            puts 'Cached repository checksum mismatch. :/ Will download.'.lightred
+          end
+        else
+          puts 'Cannot find cached repository. :/ Will download.'.lightred
+        end
+      end
+      # Download via git
+      Dir.mkdir @extract_dir
+      Dir.chdir @extract_dir do
+        system 'git init'
+        system 'git config advice.detachedHead false'
+        system 'git config init.defaultBranch master'
+        system "git remote add origin #{@pkg.source_url}", exception: true
+        system "git fetch --depth 1 origin #{@pkg.git_hashtag}", exception: true
+        system 'git checkout FETCH_HEAD'
+        system 'git submodule update --init --recursive'
+        puts "Repository downloaded".lightgreen
+      end
+      # Stow file in cache if requested
+      if CREW_CACHE_ENABLED
+        puts 'Caching downloaded archive.'.lightgreen
+          Dir.chdir "#{@extract_dir}" do system "tar c#{@verbose}Jf #{CREW_CACHE_DIR}/#{filename}.tar.xz $(find -mindepth 1 -maxdepth 1 -printf '%P\n')" end
```

Comment:
Can we split into separate lines to prevent horizontal scrolling?
```
Dir.chdir @extract_dir do
  system "tar c#{@verbose}Jf #{CREW_CACHE_DIR}/#{filename}.tar.xz $(find -mindepth 1 -maxdepth 1 -printf '%P\n')"
end
```

Reply:
Good idea.

Reply:
Changing it to 
```
        puts 'Caching downloaded archive.'.lightgreen
          Dir.chdir "#{@extract_dir}" do 
            system "tar c#{@verbose}Jf #{CREW_CACHE_DIR}/#{filename}.tar.xz \
              $(find -mindepth 1 -maxdepth 1 -printf '%P\n')"
          end
```